Docker.DotNet/Microsoft.Net.Http.Client/ManagedHandler.cs

```diff
@@ -141,7 +152,25 @@ private bool IsAllowedRedirectResponse(HttpRequestMessage request, HttpResponseM
             request.Headers.ConnectionClose = true; // TODO: Connection re-use is not supported.
 
             ProxyMode proxyMode = DetermineProxyModeAndAddressLine(request);
-            Stream transport = await ConnectAsync(request, cancellationToken);
+            Socket socket;
+            Stream transport;
+            try
+            {
+                if (_socketOpener != null)
+                {
+                    socket = await _socketOpener(request.GetConnectionHostProperty(), request.GetConnectionPortProperty().Value, cancellationToken).ConfigureAwait(false);
```

Comment:
The .Value call implies to me that its nullable. Do we always have a value? Do you want `GetValueOrDefault(80)`